docs/create-service-descriptor.md

```diff
@@ -0,0 +1,97 @@
+6. `boundingQuery` = optional JSON object that captures a serialized CTS query which will constrain all queries on 
+   this layer for features. 
+7. `geometry` = required JSON object that defines the features associated with this layer; defined further below.
+8. `dataSources` = array of JSON objects that define queryable data in MarkLogic; defined further below.
+9. `schema` = optional string; defined further below.
+10. `view` = optional string; defined further below. 
+
+The layer can contain any number of additional fields, which will be included when a client requests a service
+descriptor. For example, while GDS does not make use of an `extent` field, a user will typically want to include that
+when using the MarkLogic Koop provider so that ArcGIS clients can leverage this field.
+
+#### Data sources
+
+When defining a layer, a user has two choices for defining the source of data to be queried in MarkLogic:
+
+1. Use `schema` and `view` to identify a [TDE template](https://docs.marklogic.com/guide/app-dev/TDE). 
+2. Use the `dataSources` array.
+
+The `dataSources` array is intended to provide more flexibility by allowing for multiple sources of data that can be
+joined together. Each object in the array can have the following fields:
+
+1. `source` = required string; either `view` or `sparql`.
+2. `schema` = required when `source` is `view`; the schema associated with a TDE template.
+3. `view` = required when `source` is `view`; the view associated with a TDE template.
+4. `query` = required when `source` is `sparql`; a SPARQL query. 
+5. `joinOn` = required for the second data source; a JSON object with two keys - `left` and `right` - that 
+   identify how data from this data source should be joined with data from the first data source. The `left` key 
+   identifies a column in the first data source, while `right` identifies a column in the second data source. The 
+   object may optionally have a `joinType` key that defines the type of join; supported values are `inner` (the 
+   default), `left outer`, and `full outer`. 
+6. `joinFunction` = TODO Don't know yet. 
+7. `fields` = required JSON object when `source` is `sparql`; defines fields to add to each feature. Each key in 
+   this object is the name of an additional feature field. Each key is an object itself with a
+   `scalarType` key that identifies the type of column.
+8. `includeFields` = optional array that identifies a subset of the field names to be included in each feature. 
+9. `fragmentIdColumn` = optional string for when `source` = `view`; allows for specifying a column to be used for a 
+   fragment ID when performing a join. Specifically, this allows for passing in a value for the `systemCols` 
+   argument when GDS uses [op.fromView](https://docs.marklogic.com/op.fromView) to build an Optic pipeline based on 
+   a view. 
+
+TODO `fields` can also be used when source=view; I'm not sure why though, as it seems GDS can already determine the
```

Comment:
@jkerr5 I couldn't figure this one out - I was thinking `fields` was just for defining scalar types when using a SPARQL query, but it's used when the source is a view too.